### PR TITLE
[stable/mariadb] Feature/allow for existing pvc

### PR DESCRIPTION
**Pull request by jldeen**
**What this PR does / why we need it**:
Allows for users to add existing PVs/PVCs

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
Doesn't so much fix an issue, but does allow for users to use the chart with an existing PV/PVC
**Special notes for your reviewer**:

### CODE REVIEW

**Comment by cpanato** [comment]
@jldeen please bump the chart version and sign the DCO

**Comment by jldeen** [comment]
@cpanato Chart version bumped and DCO signed

File: stable/mariadb/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: mariadb
-version: 4.4.0
+version: 4.4.1
```
Comment (juan131) [inline stable/mariadb/Chart.yaml:2]: Increase minor version

**Review by juan131 — CHANGES_REQUESTED** [review]
Please update the README.md file accordingly to the changes you introduced.

Check Redis README.md section for existing claims for instance.

File: stable/mariadb/README.md [hunk 2]
```diff
@@ -66,6 +66,8 @@ The following table lists the configurable parameters of the MariaDB chart and t
 | `master.antiAffinity`                     | Master pod anti-affinity policy                     | `soft`                                                            |
 | `master.tolerations`                      | List of node taints to tolerate (master)            | `[]`                                                              |
 | `master.persistence.enabled`              | Enable persistence using a `PersistentVolumeClaim`  | `true`                                                            |
+| `master.persistence.existingClaim`        | Provide an existing `PersistentVolumeClaim`  | `nil`
+| `master.persistence.mountPath`            | Configure existing `PersistentVolumeClaim` mount path  | `""`
```
Comment (jldeen) [inline stable/mariadb/README.md:70]: readme updated

**Review by juan131 — APPROVED** [review]
Great! Thanks

**Comment by jldeen** [comment]
Believe the failing is related to open [issue #148](https://github.com/bitnami/bitnami-docker-mariadb/issues/148) on the bitnami/bitnami-docker-mariadb repo and is not related to the proposed changes in this PR.

File: stable/mariadb/templates/master-statefulset.yaml [hunk 3]
```diff
@@ -163,6 +167,11 @@ spec:
 {{ toYaml .Values.metrics.resources | indent 10 }}
 {{- end }}
       volumes:
+      {{- if (.Values.master.persistence.enabled) and (.Values.master.persistence.existingClaim) }}
```
Comment (juan131) [inline stable/mariadb/templates/master-statefulset.yaml]: You're not using **and** correctly. It should be:

```
{{- if and .Values.master.persistence.enabled .Values.master.persistence.existingClaim }}
```

File: stable/mariadb/templates/master-statefulset.yaml [hunk 4]
```diff
@@ -132,6 +132,10 @@ spec:
         - name: config
           mountPath: /opt/bitnami/mariadb/conf/my.cnf
           subPath: my.cnf
+{{- if .Values.master.persistence.existingClaim }}
+        - name: {{ .Values.master.persistence.existingClaim | default (include "mariadb.fullname" .) }}
```
Comment (juan131) [inline stable/mariadb/templates/master-statefulset.yaml]: Do not use defaults.

File: stable/mariadb/templates/master-statefulset.yaml [hunk 5]
```diff
@@ -163,6 +167,11 @@ spec:
 {{ toYaml .Values.metrics.resources | indent 10 }}
 {{- end }}
       volumes:
+      {{- if (.Values.master.persistence.enabled) and (.Values.master.persistence.existingClaim) }}
+        - name: {{ .Values.master.persistence.existingClaim | default (include "mariadb.fullname" .) }}
+          persistentVolumeClaim:
+            claimName: {{ .Values.master.persistence.existingClaim | default (include "mariadb.fullname" .) }}
```
Comment (juan131) [inline stable/mariadb/templates/master-statefulset.yaml]: Do not use defaults.

**Review by juan131 — CHANGES_REQUESTED** [review]
Could you please take a look to my comments?

File: stable/mariadb/templates/master-statefulset.yaml (hunk 3 above)
Comment (jldeen) [reply stable/mariadb/templates/master-statefulset.yaml]: Corrected in latest push

File: stable/mariadb/templates/master-statefulset.yaml (hunk 4 above)
Comment (jldeen) [reply stable/mariadb/templates/master-statefulset.yaml]: Corrected in latest push

File: stable/mariadb/templates/master-statefulset.yaml (hunk 5 above)
Comment (jldeen) [reply stable/mariadb/templates/master-statefulset.yaml]: Corrected in latest push

**Comment by jldeen** [comment]
Hi @juan131 

I have made the proposed changes.

**Pull request merged**